lib/Assets.js

```diff
@@ -1,96 +1,82 @@
+const asyncForEach = require('./asyncForEach.js')
+
 class Assets {
   constructor(space, { updatedContent, newContent, removedContent }) {
     this.space = space;
-    this.updatedAssets = updatedContent;
-    this.newAssets = newContent;
+    this.updatedAssets = this.filterAssets(updatedContent);
+    this.newAssets = this.filterAssets(newContent);
     this.removedAssets = removedContent;
   }
-  upsertAndDelete() {
-    return this.updateAssets()
-    .then(() => this.createAssets())
-    .then(() => this.removeAssets())
-    .catch((err) => console.error(err));
+  filterAssets(assets) {
+    return assets.filter( asset => Object.keys(asset.fields).length && asset.fields.file && (asset.fields.file && asset.fields.file['en-US'].url));
   }
-  updateAssets () {
-    const { updatedAssets, space } = this;
-    if (!updatedAssets.length) {
-      console.log('No assets to update. Moving along...')
-      return Promise.resolve();
+  async upsertAndDelete() {
+    const { updatedAssets, newAssets, removedAssets, space } = this;
+    if (updatedAssets.length) {
```

Comment:
We have this same logic in each data type's `upsertAndDelete` method. Seems ripe for refactoring.